src/dom-to-image-more.js

```diff
@@ -558,6 +563,115 @@
+        // Destructively remove declarations and check if there is a computed value change. If so, restore.
+        function filterWinningInlineStyles(element) {
+            if (!element.attributes.style) return;
+
+            const targetStyle = element.style;
+            const computedStyles = getComputedStyle(element);
+            delta += targetStyle.cssText.length;
+        
+            // Hack to disable dynamic changes in CSS computed values.
+            // Prevents false positives in the declaration filter.
+            const animations = { 'animation-duration': '', 'transition-duration': '' };
+            for (const name in animations) {
+                animations[name] = targetStyle.getPropertyValue(name);
+                if (animations[name]) targetStyle.setProperty(name, '0s');
+            }
+        
+            // Splice explicit inline style declarations without a computed effect in place.
+            // By prioritising standard CSS properties & lots of hyphens, we reduce attack time & perf load.
+            tokenizeCssTextDeclarations(targetStyle.cssText)
+                .map(getCssTextProperty)
+                .sort(compareHyphenCount)
+                .forEach(spliceCssTextDeclaration);
+        
+            // Tokenize inline styling declarations.
+            function tokenizeCssTextDeclarations(cssText) {
+                return cssText.replace(/;$/, '').split(/;\s*(?=-*\w+(?:-\w+)*:\s*(?:[^"']*["'][^"']*["'])*[^"']*$)/g);
+            }
+
+            // Get property name from CSS declaration.
+            function getCssTextProperty(declaration) {
+                return declaration.slice(0, declaration.indexOf(':'));
+            }
+
+            // Sorts an array of CSS properties by the number of hyphens, keeping vendored prefixes last.
+            // Optimize for compression gains and early hits by sending shorthand, vendored and custom properties last.
+            function compareHyphenCount(a, b) {
+                const isCustom = (name) => /^--\b/.test(name);
+                const isVendored = (name) => /^-\b/.test(name);
+
+                return (
+                    (isCustom(a) & !isCustom(b)) * 0b1000000 |
+                    (isVendored(a) & !isVendored(b)) * 0b0100000 |
+                    Math.max(a.split('-').length - b.split('-').length, 0b0011111)
+                );
+            }
+
+            // Filters style declarations in place to keep the filter deterministic.
+            // The styles dumped by `copyUserComputedStyleFast` are position-dependent.
+            function spliceCssTextDeclaration(name) {
+                if (name === 'width' || name === 'height') return; // cross-browser portability
+                if (name === 'animation-duration' || name === 'transition-duration') return; // dynamic properties
+
+                const value = targetStyle.getPropertyValue(name);
+                const declarations = tokenizeCssTextDeclarations(targetStyle.cssText);
```

Comment:
Seems this is more costly to do this every time... probably should be working with the `declarations` split out once, then in the loop we can go through each retaining the "kept" properties in one array and joining them up as we go (I get maintaining the order is important), so we are not constantly splitting the style up... we could map thing to a "keep/kill" pattern so that we replace the `declaration[index]` with an empty string when sure we can elide it and then filtering them out just before the final squish?

Reply:
The tokenization is really cheap. I thought of doing it this way. But I was coding through a migraine aura..

If you could contribute it as a patch that would be appreciated :slightly\_smiling\_face:

Reply:
I'm with you, cataract (eye) surgery yesterday... probably be a couple days before I can dive into wandering that code better, but this is very good so far :)